File: c/randomSwap.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <string.h>

void trocaLetra(char* s, char first, char second) {
    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] == first) {
            s[i] = second;
        }
    }
}

int stringEq(char* testado, char* testando) {
    while (*testado && *testando) {
        if (*testado != *testando) {
            return 0;
        }
        testado++;
        testando++;
    }
    return (*testado == '\0' && *testando == '\0');
}

void swapAndPrint() {
    char *input = NULL;
    size_t buffer_size = 256;

    getline(&input, &buffer_size, stdin);

    if (input[strlen(input) - 1] == '\n') { // rm quebra de linha
        input[strlen(input) - 1] = '\0';
    }

    if (stringEq(input, "FIM")) {
        free(input);
        return;
    }

    char firstLetter = 'a' + rand() % 26;
    char secondLetter = 'a' + rand() % 26;

    trocaLetra(input, firstLetter, secondLetter);
    printf("%s\n", input);

    free(input);
    swapAndPrint(); // Recursivo
}

int main() {
    srand(4);

    swapAndPrint();

    return 0;
}

File: RandomSwap.java
import java.util.Random;
import java.util.Scanner;

public class RandomSwap {
    public static void main(String[] args) {
        Random gerador = new Random();
        gerador.setSeed(4);
        Scanner scan = new Scanner(System.in);
        String input = scan.nextLine();
        
        while (!stringEq(input, "FIM")) {
            char firstLetter = (char) ('a' + (Math.abs(gerador.nextInt()) % 26));
            char secondLetter = (char) ('a' + (Math.abs(gerador.nextInt()) % 26));
            String result = trocaLetra(input, firstLetter, secondLetter);
            System.out.println(result);
            
            input = scan.nextLine();
        }
        
        scan.close();
    }

    public static String trocaLetra(String s, char first, char second) {
        String result = "";
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == first) {
                result += second;
            } else {
                result += s.charAt(i);
            }
        }
        return result;
    }

    public static boolean stringEq(String testado, String testando) {
        for (int i = 0; i < testado.length(); i++) {
            if (testado.charAt(i) != testando.charAt(i))
                return false;
        }
        return true;
    }
}